res/controllers/Denon-MC7000-scripts.js

```diff
@@ -669,33 +729,33 @@ MC7000.nextRateRange = function(midichan, control, value, status, group) {
     if (value === 0) {
         return; // don't respond to note off messages
     }
-    var deckNumber = script.deckFromGroup(group);
+    var deckOffset = script.deckFromGroup(group) - 1;
     // increment currentRateRangeIndex and check for overflow
-    if (++MC7000.currentRateRangeIndex[deckNumber - 1] ===
+    if (++MC7000.currentRateRangeIndex[deckOffset] ===
       MC7000.rateRanges.length) {
-        MC7000.currentRateRangeIndex[deckNumber - 1] = 0;
+        MC7000.currentRateRangeIndex[deckOffset] = 0;
     }
-    engine.setValue(group, "rateRange", MC7000.rateRanges[MC7000.currentRateRangeIndex[deckNumber - 1]]);
+    engine.setValue(group, "rateRange", MC7000.rateRanges[MC7000.currentRateRangeIndex[deckOffset]]);
 };
 
 // Previous Rate range toggle
 MC7000.prevRateRange = function(midichan, control, value, status, group) {
     if (value === 0) {
         return; // don't respond to note off messages
     }
-    var deckNumber = script.deckFromGroup(group);
+    var deckOffset = script.deckFromGroup(group) - 1;
```

Comment:
`script.deckFromGroup(group) - 1` is used very often. You might consider to extract a tiny helper function `deckOffsetFromGroup()` to not forget about the `- 1`.

Reply:
Thanks Uwe - it gets strange if 3 people comment to the code review. I followed more or less and changed all the variables regarding deck or offset number [as suggested here](https://github.com/mixxxdj/mixxx/pull/3738#discussion_r599365038) and mainly I used `deckNumber - 1` but if i didn't need `deckNumber` then only `deckOffset` is used. 

At the end it is the same and 3 people had 3 different opinions which gets a little frustrating.

@uklotzde , @Swiftb0y , @ronso0 ... Can you please clarify internally and then let me know which way you wanna go and I possibly change all over again.

Thanks you.

Reply:
Using deckOffset is fine. I am just suggesting to move the duplicated code into a small helper function and then use that instead of the script.deckFromGroup() followed by the subtraction. The code will get more readable, no functional change.

Reply:
Just some final polishing. Actually I wanted to ask if this PR is ready for merge before I stumbled upon this simple and obvious improvement. One of the basic [refactoring patterns](https://wiki.c2.com/?ExtractMethod).

Reply:
I agree, the change is not really conflicting with everything we discussed so far. Its just an addition to extract the `script.deckFromGroup(group) - 1` part into its own function. this way its less likely you may forget the `- 1`part in the future.

Reply:
I'd say we should merge now, the change can go in a separate PR.